doc/source/cluster/kubernetes/examples/distributed-checkpointing-with-gcsfuse.md

```diff
@@ -0,0 +1,296 @@
+(kuberay-distributed-checkpointing-gcsefuse)=
+
+# Distributed Checkpointing with KubeRay and GCSFuse
+
+This example demonstrates how to orchestrate distributed checkpointing with KubeRay, using the GCSFuse CSI driver and Google Cloud Storage as the remote storage system.
+To illustrate the concepts, this guide uses the [Finetuning a Pytorch Image Classifier with Ray Train](https://docs.ray.io/en/latest/train/examples/pytorch/pytorch_resnet_finetune.html) example.
+
+## Why Distributed Checkpointing with GCSFuse?
+
+In large-scale, high-performance machine learning, distributed checkpointing is crucial for fault tolerance, ensuring that if a node fails during training, the process can be resumed from the latest saved checkpoint instead of starting from scratch.
```

Comment:
```suggestion
In large-scale, high-performance machine learning, distributed checkpointing is crucial for fault tolerance, ensuring that if a node fails during training, Ray can resume the process from the latest saved checkpoint instead of starting from scratch.
```